connectors/src/connectors/zendesk/temporal/sync_ticket.ts

```diff
@@ -17,59 +43,157 @@ export async function syncTicket({
+    const metadata = [
+      `Ticket ID: ${ticket.id}`,
+      `Subject: ${ticket.subject}`,
+      `Created At: ${createdAtDate.toISOString()}`,
+      `Updated At: ${updatedAtDate.toISOString()}`,
+      `Status: ${ticket.status}`,
+      `Priority: ${ticket.priority || "N/A"}`,
+      `Type: ${ticket.type || "N/A"}`,
+      `Channel: ${ticket.via.channel}`,
+      `Requester: ${ticket.requester.name} (${ticket.requester.email})`,
+      ticket.assignee_id ? `Assignee ID: ${ticket.assignee_id}` : "Unassigned",
+      `Organization ID: ${ticket.organization_id || "N/A"}`,
+      `Group ID: ${ticket.group_id || "N/A"}`,
+      `Tags: ${ticket.tags.length ? ticket.tags.join(", ") : "No tags"}`,
+      ticket.due_at
+        ? `Due Date: ${new Date(ticket.due_at).toISOString()}`
+        : null,
+      ticket.satisfaction_rating
+        ? `Satisfaction Rating: ${ticket.satisfaction_rating.score}`
+        : null,
+      ticket.satisfaction_rating?.comment
+        ? `Satisfaction Comment: ${ticket.satisfaction_rating.comment}`
+        : null,
+      `Has Incidents: ${ticket.has_incidents ? "Yes" : "No"}`,
+      ticket.problem_id ? `Related Problem ID: ${ticket.problem_id}` : null,
+      `Custom Fields:`,
+      ...ticket.custom_fields.map(
+        (field) => `  - Field ${field.id}: ${field.value || "N/A"}`
+      ),
+    ]
+      .filter(Boolean)
+      .join("\n");
+
+    const ticketContent = `
+${metadata}
+
+Initial Description:
+${ticket.description || "No initial description"}
+
+Conversation:
+${comments
+  .map((comment) => {
+    const author = users.get(comment.author_id);
+    return `
+[${new Date(comment.created_at).toISOString()}] ${
+      author ? `${author.name} (${author.email})` : "Unknown User"
+    }:
+${comment.body}`;
+  })
+  .join("\n")}
+`.trim();
+
+    const ticketContentInMarkdown = turndownService.turndown(ticketContent);
+
+    const renderedMarkdown = await renderMarkdownSection(
+      dataSourceConfig,
+      ticketContentInMarkdown
+    );
+    const documentContent = await renderDocumentTitleAndContent({
+      dataSourceConfig,
+      title: ticket.subject,
+      content: renderedMarkdown,
       createdAt: createdAtDate,
       updatedAt: updatedAtDate,
-      assigneeId: ticket.assignee_id,
-      groupId: ticket.group_id,
-      organizationId: ticket.organization_id,
-      description: ticket.description,
-      subject: ticket.subject,
-      requesterMail: ticket?.requester?.email || "",
-      url: ticket.url,
-      satisfactionScore: ticket.satisfaction_rating?.score || "unknown",
-      satisfactionComment: ticket.satisfaction_rating?.comment || "",
-      status: ticket.status,
+    });
+
+    const documentId = getTicketInternalId(connectorId, ticket.id);
+
+    await upsertToDatasource({
+      dataSourceConfig,
+      documentId,
+      documentContent,
+      documentUrl: ticket.url,
+      timestampMs: updatedAtDate.getTime(),
       tags: ticket.tags,
```

Comment:
We can add the title, creation date and update date to the tags as well

Reply:
Why would we need a date in tags? Tags are for me repeatable across several articles, and dates are not. Or maybe I'm seeing tags the wrong way 🤔

Reply:
we put these 2 dates in the tags pretty much everywhere, so I guess this will be expected here as well

Reply:
don't have full context regarding how tags are used, sorry for not having a better answer

Reply:
Ok, me neither haha, will add

Reply:
thanks! 🙏